src/test/fulcro/server_render_spec.cljc

```diff
@@ -46,21 +46,55 @@
    (specification "SSR script tag generation"
      (assertions
        "puts an assignment on the document window"
-       (ssr/initial-state->script-tag []) => "<script type='text/javascript'>\nwindow.INITIAL_APP_STATE = '[]'\n</script>\n")))
+       (ssr/initial-state->script-tag []) => "<script type='text/javascript'>\nwindow.INITIAL_APP_STATE = 'W10='\n</script>\n")))
 
 #?(:clj
    (specification "SSR script tag generation w/ apos"
+                  (assertions
+                    "puts an assignment on the document window with some tricky content"
+                    (ssr/initial-state->script-tag {:some-field "some text's apostrophe"}) => "<script type='text/javascript'>\nwindow.INITIAL_APP_STATE = 'WyJeICIsIn46c29tZS1maWVsZCIsInNvbWUgdGV4dCdzIGFwb3N0cm9waGUiXQ=='\n</script>\n")))
+
+#?(:clj
+   (specification "SSR script tag generation w/ emoji"
      (assertions
-       "puts an assignment on the document window with some tricky content"
-       (ssr/initial-state->script-tag {:some-field "some text's apostrophe"}) => "<script type='text/javascript'>\nwindow.INITIAL_APP_STATE = '[\"^ \",\"~:some-field\",\"some text\\'s apostrophe\"]'\n</script>\n")))
+       "puts an assignment on the document window with some emoji content"
+       (ssr/initial-state->script-tag {:emoji "\u26d1"}) => "<script type='text/javascript'>\nwindow.INITIAL_APP_STATE = 'WyJeICIsIn46ZW1vamkiLCLim5EiXQ=='\n</script>\n")))
+
+#?(:clj
+   (specification "SSR script tag generation w/ naughty strings"
+                  (assertions
+                    "puts an assignment on the document window with some nefarious content"
+                    (ssr/initial-state->script-tag {:naughty "\u0001\u0002\u0003\u0004\u0005"}) => "<script type='text/javascript'>\nwindow.INITIAL_APP_STATE = 'WyJeICIsIn46bmF1Z2h0eSIsIlx1MDAwMVx1MDAwMlx1MDAwM1x1MDAwNFx1MDAwNSJd'\n</script>\n")))
 
 #?(:cljs
    (specification "SSR initial-state extraction"
-     (let [state (util/transit-clj->str [])]
-       (set! (.-INITIAL_APP_STATE js/window) state)
-       (assertions
-         "Can pull the initial state from the document's window"
-         (ssr/get-SSR-initial-state) => []))))
+     (set! (.-INITIAL_APP_STATE js/window) "W10=")
```

Comment:
Oh duh...That's my oversight...I guess I did write OK tests, since you're right there is no easy way to do it other than to have a test that "proves" it encodes X -> Y, and then in cljs can decode Y from X. That is the best you can do.